### DISCUSSION TITLE
PHP 7 required modules to run OpenMage

### DISCUSSION THREAD

**Discussion by addison74** [OpenMage/magento-lts#1669] [Ideas]
There are a lot of articles on Internet about what PHP modules needs Magento to run properly. Most of these articles are outdated referring to modules which are not available anymore starting with version 7.x (e.g. hash, iconv, pcre, pdo, pdo_mysql, simplexml).

**Here is my list**

```
php7.x-cgi - server-side, HTML-embedded scripting language (CGI binary)
php7.x-cli - command-line interpreter for the PHP scripting language
php7.x-common - documentation, examples and common module for PHP
php7.x-curl - CURL module for PHP
php7.x-fpm - server-side, HTML-embedded scripting language (FPM-CGI binary)
php7.x-gd - GD module for PHP
php7.x-json - JSON module for PHP
php7.x-mbstring - MBSTRING module for PHP
php7.x-mcrypt - libmcrypt module for PHP
php7.x-mysql -  MySQL module for PHP
php7.x-opcache - Zend OpCache module for PHP
php7.x-readline - readline module for PHP
php7.x-soap - SOAP module for PHP
php7.x-xml - DOM, SimpleXML, WDDX, XML, and XSL module for PHP
php7.x-zip - Zip module for PHP
```

**IMPORTANT**: Starting with PHP 7.2+ mcrypt needs to be disabled (in Linux command phpdismod php7.x-mcrypt or by editing php.ini file).

Maybe the home page of OpenMage project on GitHub should be updated with this information. If someone installed over the years Magento he knows that in installation process the script could request some PHP modules like gd, curl. If the current installation is moved to a new server and these modules are not installed before Magento won't run properly.

**Comment by colinmollenhour** [comment]
These could be added to composer.json, like `"ext-gd": "*"`. As composer is the defacto way to install I think that would cover it for most people. It could go on the "install" page and/or the README.md but I don't think it belong on the homepage.
